content/docs/rfcs/44/README.md

```diff
@@ -0,0 +1,261 @@
+---
+slug: 44
+title: 44/WAKU2-PRIVATE-SETTLEMENT
+name: Waku v2 Private Settlement
+status: draft
+tags: waku-core
+editor: Sanaz Taheri <sanaz@status.im>
+contributors:
+---
+
+Waku nodes provide services that have a specific cost and this should be incentivized in a service network. In the current specification, we present a private settlement protocol targeted for Waku v2 store protocol to incentivize nodes for their supplied storage and resolving historical queries.
+
+
+## Terminology
+The term Personally identifiable information (PII) refers to any piece of data that can be used to uniquely identify a user. 
+For example, the signature verification key, and the hash of one's static IP address are unique for each user and hence count as PII.
+
+
+# Security Objectives
+The private settlement protocol is designed to be privacy-preserving
+- The payment does not rely on direct connection between the service provider and the requester.
+- Protocol messages do not contain PII information about the requester.
+- Due to items above, the private settlement can be used in conjunction with anonymous network protocols like Tor and MixNet to additionally feature IP protection for the service requester.  
+<!-- - It hides the number and size of requests resolved by a specific store node, hence provides privacy for the service provider too. 
+Access to such information would enable inference about resources, on-line time, and wealth of a service provider hence leading to targeted attacks like DDoS. -->
+
+# Overview
+Service nodes who wish to be paid for their supplied services need to register their Ethereum address in the Waku Service Contract and also pay the membership fee `membership_fee`.  
+This fee is locked in the pool and is non refundable.
+
+Querying nodes/ service consumers who wish to pay for their received services, deposit some fund in the Waku Service Contract and obtain Service Credentials or Tokens.
+Tokens have arbitrary values which is decided by the service consumer.
+
+Upon receiving a history query, the service provider measures the cost of the service e.g., in terms of the size of the history response.
+It then communicates with the querying node 1) the cost in Ether, 2) a random `request_ID` 3) the necessary information for the payment e.g., its registered Ethereum address.
+The querying node sends a [transaction](#interaction-with-the-waku-service-contract), with the `request_ID` attached to it, to the contract to spend one of its service credential to pay to the service node.
```

Comment:
I'm confused by this part. Isn't the whole point of this design to decouple payment with service credentials from settlement? If you require contract interactions every history quest this cost gas and it makes the system not scalable due to reasonable tps limits of underlying chain. It is also not realistic in terms of latency from HistoryQuery to HistoryResponse. What am I missing?

Reply:
Or is the idea that you pre-pay for many HistoryQueries in advance? Not clear from text

Reply:
@s1fr0 both it possible, however, the first one is only possible if the spent token contains the exact amount as the amount requested by the store node (but not more). Otherwise, the token owner needs to wait for the store node to send the transaction to the contract so that it can use the remaining amount if its token.